src/main/java/gregtech/api/recipes/ingredients/GTRecipeFluidInput.java

```diff
@@ -67,30 +68,30 @@ public boolean acceptsFluid(@Nullable FluidStack input) {
         return false;
     }
 
+    @Override
+    public int hashCode() {
+        if (nbtMatcher == null) {
+            //the Fluid registered to the fluidName on game load might not be the same Fluid after loading the world, but will still have the same fluidName.
+            return Objects.hash(inputStack.getFluid().getName(), this.amount, this.nbtMatcher, this.nbtCondition, inputStack.tag);
+        }
+        return Objects.hash(inputStack.getFluid().getName(), this.amount, this.nbtMatcher, this.nbtCondition, 0);
```

Comment:
It would be better to pre-compute the hashcode and store it, in order to avoid object creation in the hashcode method, since this object will be hashed often.